include/xtypes/DynamicData.hpp

```diff
@@ -235,6 +235,173 @@ class ReadableDynamicDataRef
         catch(bool value) { return value; }
     }
 
+    /// \brief Class used for iterate ReadableDynamicDataRef
+    class Iterator
+    {
+    public:
+        Iterator(
+                const ReadableDynamicDataRef& ref)
+            : Iterator(ref, false)
+        {}
+
+        Iterator& operator = (
+                const Iterator& other)
+        {
+            data_ = other.data_;
+            index_ = other.index_;
+            return *this;
+        }
+
+        bool operator == (
+                const Iterator& other) const
+        {
+            return other.data_ == data_ && other.index_ == index_;
+        }
+
+        bool operator != (
+                const Iterator& other) const
+        {
+            return other.data_ != data_ || other.index_ != index_;
```

Comment:
```suggestion
            return !(*this == other)
```
In order to improve the maintenance